rotkehlchen/db/upgrades/v31_v32.py

```diff
@@ -86,6 +87,52 @@ def _add_new_tables(cursor: 'Cursor') -> None:
 );""")  # noqa: E501
 
 
+def _update_history_entries_from_kraken(cursor: 'Cursor') -> None:
+    """The logic for kraken was adding additional entries for trades when fee + kfee was
+    being used. This function makes the state of the database consistent with the upgraded
+    logic by:
+    - Removing extra row additions
+    - Updating the sequence index to avoid duplicates
+    - Make sure that no other event has duplicated sequence indexes
+    """
+    cursor.execute("""
+    DELETE FROM history_events where location="B" AND asset="KFEE" AND
+     type="trade" AND subtype=NULL;
+    """)
+    cursor.execute("""
+    UPDATE history_events SET sequence_index = sequence_index + 1 WHERE asset="KFEE"
+     AND location="B" AND type="trade" AND subtype="fee"
+    """)
+
+    cursor.execute("""
+    SELECT e.event_identifier, e.sequence_index, e.identifier from history_events e JOIN (SELECT event_identifier,
+    sequence_index, COUNT(*) as cnt FROM history_events GROUP BY event_identifier, sequence_index)
+    other ON e.event_identifier = other.event_identifier and e.sequence_index=other.sequence_index
+    WHERE other.cnt > 1;
+    """)  # noqa: E501
+
+    update_sentences = []
+    used_indexes: Dict[str, Set[int]] = defaultdict(set)
+    for event_identifier, sequence_index, identifier in cursor:
+        last_index = used_indexes.get(event_identifier)
+        if last_index is None:
```

Comment:
This confused me so much! The name should be plural. It's a set not an int you are getting here.

```suggestion
        last_indices = used_indexes.get(event_identifier)
        if last_indices is None:
```